File: app/src/main/java/com/example/sy43/db/DAO/SubCategoryDAO.java
package com.example.sy43.db.DAO;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;


import com.example.sy43.db.entity.SubCategory;
import com.google.common.util.concurrent.ListenableFuture;

import java.util.List;
import androidx.room.Dao;

@Dao
public interface SubCategoryDAO{
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public ListenableFuture<Void> insert(SubCategory... subCategory);

    @Query("SELECT * FROM SubCategory WHERE SubCatID LIKE :SubCatID")
    public ListenableFuture<List<SubCategory>> findByID(int SubCatID);

    @Query("SELECT CurrentValue FROM SubCategory")
    public ListenableFuture<List<Float>> findValues();

    @Query("SELECT MaxValue FROM SubCategory")
    public ListenableFuture<List<Float>> findMaxValues();

    @Query("SELECT * FROM SubCategory WHERE Category LIKE :category")
    public ListenableFuture<List<SubCategory>> findByCategoryID(int category);

    @Query("DELETE FROM subcategory WHERE SubCatID = :id")
    public ListenableFuture<Void> delSubByID(int id);

    @Delete
    void deleteSubCategory(SubCategory Subcategory);
}

File: Fluz/app/src/main/java/com/example/fluz/data/dao/SubCategoryDao.kt
package com.example.fluz.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import com.example.fluz.data.entities.Category
import com.example.fluz.data.entities.SubCategory

/**
 * SQL Requests for SubCategory entity
 */
@Dao
interface SubCategoryDao {

    /**
     * Insert a new subcategory
     *
     * @param subCategory the subcategory to insert
     */
    @Insert(onConflict = OnConflictStrategy.ABORT)
    suspend fun insert(subCategory: SubCategory)
}